### DISCUSSION TITLE
How to create Model classes?

### DISCUSSION THREAD

**Discussion by jordiblanchcarles** [Intechnity-com/OdooJsonRpcClient#49] [Q&A]
Hello,
maybe this should sound as a stupid question, but I cannot understand how to work with Model classes.

As I understand, we must query our Odoo API to get the any model's structure. I can do this following the README example, but what I get is a huge string full of "\r\n" and "<br>"....

Are we supposed to create a small command line program to ask our Odoo API each of the models we need to use and then convert these strings we are getting into coded C# classes to use in our main project?

Thank you!

**Comment by patricoos** [comment]
models are different depending on odoo version and also on installed extensions. the mapping method will return you a string with the converted odoo (python) type to .net. Using this models you won't get weird errors with types etc. 

yes, you can do it e.g. in a console application while debugging
